dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1773,4 +1826,668 @@ public void getDirectEpersonGroups() throws Exception {
+            "\"eperson.lastname\":[{\"value\":\"Doe\"}]},\"password\":\"somePassword\"," +
+            "\"type\":\"eperson\"}";
+
+        MvcResult mvcResult = getClient().perform(post("/api/eperson/epersons")
+                                .param("token", newRegisterToken)
+                                .content(json)
+                                .contentType(MediaType.APPLICATION_JSON))
+                                  .andExpect(status().isCreated())
+                                  .andExpect(jsonPath("$", Matchers.allOf(
+                            hasJsonPath("$.uuid", not(empty())),
+                            // is it what you expect? EPerson.getName() returns the email...
+                            //hasJsonPath("$.name", is("Doe John")),
+                            hasJsonPath("$.type", is("eperson")),
+                            hasJsonPath("$._links.self.href", not(empty())),
+                            hasJsonPath("$.metadata", Matchers.allOf(
+                                matchMetadata("eperson.firstname", "John"),
+                                matchMetadata("eperson.lastname", "Doe")
+                            ))))).andReturn();
+
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String epersonUuid = String.valueOf(map.get("uuid"));
+        EPerson createdEPerson = ePersonService.find(context, UUID.fromString(epersonUuid));
+        assertTrue(ePersonService.checkPassword(context, createdEPerson, "somePassword"));
+
+        assertNull(registrationDataService.findByToken(context, newRegisterToken));
+        context.turnOffAuthorisationSystem();
+        ePersonService.delete(context, createdEPerson);
+        context.restoreAuthSystemState();
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.deleteByToken(context, newRegisterToken);
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void postEPersonWithTokenWithEmailProperty() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        String newRegisterEmail = "new-register@fake-email.com";
+        RegistrationRest registrationRest = new RegistrationRest();
+        registrationRest.setEmail(newRegisterEmail);
+        getClient().perform(post("/api/eperson/registrations")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(mapper.writeValueAsBytes(registrationRest)))
+                   .andExpect(status().isCreated());
+        String newRegisterToken = registrationDataService.findByEmail(context, newRegisterEmail).getToken();
+
+        String json = "{\"metadata\":{\"eperson.firstname\":[{\"value\":\"John\"}]," +
+            "\"eperson.lastname\":[{\"value\":\"Doe\"}]},\"email\":\"" + newRegisterEmail +
+            "\",\"password\":\"somePassword\",\"type\":\"eperson\"}";
```

Comment:
Again, no manually created JSON please

Reply:
The reason why this is necessary for this test has been documented at https://github.com/DSpace/DSpace/pull/2763/commits/4ea4f9112948d28858fb9b8f727da75fe4312032#diff-a256d3cb30e4b4ad847136e8ffedb15aR2119